src/coreclr/vm/classlayoutinfo.cpp

```diff
@@ -328,6 +328,20 @@ namespace
         return FALSE;
     }
 
+    BOOL TypeHasInt128Field(CorElementType corElemType, TypeHandle pNestedType)
+    {
+        if (CorTypeInfo::IsPrimitiveType(corElemType) || corElemType == ELEMENT_TYPE_PTR || corElemType == ELEMENT_TYPE_FNPTR)
+        {
+            return FALSE;
+        }
```

Comment:
This block is unneeded since we only return true in the `ELEMENT_TYPE_VALUETYPE` branch.

```suggestion
```